Source/WebCore/cssjit/SelectorCompiler.cpp

```diff
@@ -4186,11 +4187,17 @@ void SelectorCodeGenerator::generateElementIsScopeRoot(Assembler::JumpList& fail
     failureCases.append(m_assembler.branchPtr(Assembler::NotEqual, scope, elementAddressRegister));
 }
 
+JSC_DEFINE_JIT_OPERATION(operationElementIsTarget, bool, (const Element* element))
+{
+    return element == element->document().cssTarget() || InspectorInstrumentation::forcePseudoState(*element, CSSSelector::PseudoClassTarget);
+}
+
 void SelectorCodeGenerator::generateElementIsTarget(Assembler::JumpList& failureCases)
 {
-    LocalRegister document(m_registerAllocator);
-    DOMJIT::loadDocument(m_assembler, elementAddressRegister, document);
-    failureCases.append(m_assembler.branchPtr(Assembler::NotEqual, Assembler::Address(document, Document::cssTargetMemoryOffset()), elementAddressRegister));
```

Comment:
Is there any way to access to Inspector instrumentation information from JIT? I think this change is regressing target pseudo selector performance, so if we have a way to inline things without calling a function, that's great.

Reply:
I kinda doubt it.  `InspectorInstrumentation` is a somewhat complex passthrough layer that (eventually) calls into `InspectorCSSAgent`, which then does some `HashMap` and `HashSet` lookups for the given `Element` and `CSSSelector::PseudoClassType`.  I'm all ears though if you have an idea about how to make `InspectorInstrumentation` work from JIT :)

FWIW we already have this behavior/logic with `:hover:`, `:active`, etc. so I almost wanna argue that there's precedent for this.

To take a step back, is it actually necessary for me to change the CSS JIT at all in order for this to work (i.e. are the changes in `Source/WebCore/css/SelectorChecker.cpp` enough)?  Or is the CSS JIT also used for style resolution, etc.?

Reply:
CSS JUT is used for style checking. Thus, SelectoeChecker change needs ti be in sync with CSS JIT.